src/distributed_graph_generator.cc

```diff
@@ -402,6 +423,29 @@ void distributed_graph_generator::generate_distributed_commands(const task& tsk)
 		}
 	}
 
+	// Check for and report overlapping writes between local chunks, and between local and remote chunks.
```

Comment:
I don't know how cumbersome the signature would be, but if the answer is "not *too* cumbersome", then I'd say this entire `if` is a good candidate for moving to a separate (member?) function (out of an already large one).

Reply:
Done.